docker/autoware-universe/Dockerfile

```
@@ -53,14 +53,70 @@ RUN chmod 644 /etc/glvnd/egl_vendor.d/10_nvidia.json
 RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash" > /etc/bash.bashrc
 CMD ["/bin/bash"]
 
-FROM devel as prebuilt
+FROM devel as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ## Build and change permission for runtime data conversion
 RUN source /opt/ros/"$ROS_DISTRO"/setup.bash \
   && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release \
   && find /autoware/install -type d -exec chmod 777 {} \;
 
+ARG CUDA_RUNTIME_BASE_IMAGE
+# hadolint ignore=DL3006
+FROM $CUDA_RUNTIME_BASE_IMAGE as prebuilt
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ARG ROS_DISTRO
+ARG SETUP_ARGS
+
+## Install apt packages
+# hadolint ignore=DL3008
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+  git \
+  ssh \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+## Copy files
+COPY autoware.repos setup-dev-env.sh ansible-galaxy-requirements.yaml amd64.env arm64.env /autoware/
+COPY ansible/ /autoware/ansible/
+WORKDIR /autoware
+RUN ls /autoware
+
+## Add GitHub to known hosts for private repositories
+RUN mkdir -p ~/.ssh \
+  && ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+## Set up runtime environment
+RUN --mount=type=ssh \
+  ./setup-dev-env.sh -y $SETUP_ARGS --no-cuda-drivers --no-dev --ros-install-base universe \
```

Is this option valid?

```suggestion
  ./setup-dev-env.sh -y $SETUP_ARGS --no-cuda-drivers --no-dev universe \
```

---

Also, I guess `--no-cuda-drivers` should come from `SETUP_ARGS`.

https://github.com/autowarefoundation/autoware/blob/b64cfdb55c2106934b5262abd8667c65671ed597/.github/workflows/docker-build-and-push-main.yaml#L27

```suggestion
  ./setup-dev-env.sh -y $SETUP_ARGS --no-dev universe \
```

---

`--ros-install-base` It was valid before, but I changed the code, so need to be removed. I will add into $SETUP_ARGS